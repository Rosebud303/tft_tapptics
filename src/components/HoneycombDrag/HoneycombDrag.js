import React, { Component } from 'react'
import { HexUtils, GridGenerator, Layout, Hexagon, Text, Pattern } from 'react-hexgrid';

class HoneycombDrag extends Component {


  render() {
    // const { hexagons
    let champs = this.props.champions
    let hexagons = GridGenerator.parallelogram(-2, 3, -2, 1).map(hexagon => {
      return champs.map(champ => {
        let name = champ.attributes.data.name.split(',')
        return Object.assign({}, hexagon, {
          text: name[0],
          image: champ.attributes.data.champion_thumbnail
        });
      })
    })

    return (
         <Layout className="HoneycombDrag" size={{ x: 5, y: 5 }} flat={false} spacing={1.05} origin={{ x: -40, y: 50 }}>
        {
          hexagons[0].map((hex, i) => (
            <Hexagon
              key={i}
              q={hex.q+i}
              r={hex.r}
              s={hex.s}
              fill={hex.image ? HexUtils.getID(hex) : null}
              data={hex}
              // onDragStart={(e, h) => this.onDragStart(e, h)}
              // onDragEnd={(e, h, s) => this.onDragEnd(e, h, s)}
            >
              <Text className="hex-text">{hex.text}</Text>
              { hex.image && <Pattern id={HexUtils.getID(hex)} link={hex.image} /> }
              {/* { console.log(this.props.champions.find(champ => champ.attributes.data.name.includes(hex.text).attributes.data.champion_thumbnail))} */}
            </Hexagon>
          ))
        }
      </Layout>
    )
  }
}

export default HoneycombDrag;