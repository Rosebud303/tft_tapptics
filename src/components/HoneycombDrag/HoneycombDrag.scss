.tiles {
  margin: 2px;
  display: inline-block;
  height: 45px;
  width: 40px;
  clip-path: polygon(50% 0%, 100% 25%, 100% 75%, 50% 100%, 0% 75%, 0% 25%);
  position: relative;
}

.icon {
  position: absolute;
  height: 45px;
  width: 40px;
}