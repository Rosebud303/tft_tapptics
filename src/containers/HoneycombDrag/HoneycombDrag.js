import React, { Component } from 'react'
import { HexGrid, Layout, Hexagon, Text, Pattern, Path, Hex } from 'react-hexgrid';

export default class HoneycombDrag extends Component {
  render() {
    return (
      <section className="HoneycombDrag">
        <HexGrid width={1000} height={250} viewBox="-50 -50 100 100">
          <Layout size={{ x: 8, y: 8 }} flat={false} spacing={1.1} origin={{ x: 0, y: 0 }}>
            <Hexagon q={-6} r={-2} s={0}  />
            <Hexagon q={-5} r={-2} s={0}  />
            <Hexagon q={-4} r={-2} s={0}  />
            <Hexagon q={-3} r={-2} s={0}  />
            <Hexagon q={-2} r={-2} s={0}  />
            <Hexagon q={-1} r={-2} s={0}  />
            <Hexagon q={0} r={-2} s={0}  />
            <Hexagon q={1} r={-2} s={0}  />
            <Hexagon q={2} r={-2} s={0}  />
            <Hexagon q={3} r={-2} s={0}  />
            <Hexagon q={4} r={-2} s={0}  />
            <Hexagon q={5} r={-2} s={0}  />
            <Hexagon q={6} r={-2} s={0}  />
            <Hexagon q={-6} r={-1} s={0}  />
            <Hexagon q={-5} r={-1} s={0}  />
            <Hexagon q={-4} r={-1} s={0}  />
            <Hexagon q={-3} r={-1} s={0}  />
            <Hexagon q={-2} r={-1} s={0}  />
            <Hexagon q={-1} r={-1} s={0}  />
            <Hexagon q={0} r={-1} s={0}  />
            <Hexagon q={1} r={-1} s={0}  />
            <Hexagon q={2} r={-1} s={0}  />
            <Hexagon q={3} r={-1} s={0}  />
            <Hexagon q={4} r={-1} s={0}  />
            <Hexagon q={5} r={-1} s={0}  />
            <Hexagon q={-7} r={0} s={0}  />
            <Hexagon q={-6} r={0} s={0}  />
            <Hexagon q={-5} r={0} s={0}  />
            <Hexagon q={-4} r={0} s={0}  />
            <Hexagon q={-3} r={0} s={0}  />
            <Hexagon q={-2} r={0} s={0}  />
            <Hexagon q={-1} r={0} s={0}  />
            <Hexagon q={0} r={0} s={0}  />
            <Hexagon q={1} r={0} s={0}  />
            <Hexagon q={2} r={0} s={0}  />
            <Hexagon q={3} r={0} s={0}  />
            <Hexagon q={4} r={0} s={0}  />
            <Hexagon q={5} r={0} s={0}  />
            <Hexagon q={-7} r={1} s={0}  />
            <Hexagon q={-6} r={1} s={0}  />
            <Hexagon q={-5} r={1} s={0}  />
            <Hexagon q={-4} r={1} s={0}  />
            <Hexagon q={-3} r={1} s={0}  />
            <Hexagon q={-2} r={1} s={0}  />
            <Hexagon q={-1} r={1} s={0}  />
            <Hexagon q={0} r={1} s={0}  />
            <Hexagon q={1} r={1} s={0}  />
            <Hexagon q={2} r={1} s={0}  />
            <Hexagon q={3} r={1} s={0}  />
            <Hexagon q={4} r={1} s={0}  />
            <Path start={new Hex(0, 0, 0)} end={new Hex(-2, 0, 1)} />
          </Layout>
        </HexGrid>
      </section>
    )
  }
}
