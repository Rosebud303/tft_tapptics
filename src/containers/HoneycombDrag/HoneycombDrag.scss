.HoneycombDrag {
  
  svg polygon {
    fill: $overlay;
    stroke: $cell-border;
    stroke-width: .5;
  }
}