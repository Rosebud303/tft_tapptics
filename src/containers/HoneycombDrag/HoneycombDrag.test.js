import React from 'react';
import { shallow } from 'enzyme';
import HoneycombDrag from './HoneycombDrag';

describe('HoneycombDrag', () => {

  let wrapper;
  let champs = [{attributes: { data: { name: 'Vayne' } } }]

  beforeEach(() =>{
    wrapper = shallow(<HoneycombDrag champions={champs} />);
  });

  it('should match the snapshot with the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});