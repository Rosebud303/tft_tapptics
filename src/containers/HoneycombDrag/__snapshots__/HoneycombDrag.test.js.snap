// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoneycombDrag should match the snapshot with the data passed in 1`] = `
<section
  className="HoneycombDrag"
>
  <HexGrid
    height={250}
    viewBox="-50 -50 100 100"
    width={1000}
  >
    <Layout
      flat={false}
      origin={
        Object {
          "x": -10,
          "y": -10,
        }
      }
      size={
        Object {
          "x": 9,
          "y": 9,
        }
      }
      spacing={1.1}
    >
      <Hexagon
        q={-6}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-5}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-4}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-3}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-2}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-1}
        r={-2}
        s={0}
      />
      <Hexagon
        q={0}
        r={-2}
        s={0}
      />
      <Hexagon
        q={1}
        r={-2}
        s={0}
      />
      <Hexagon
        q={2}
        r={-2}
        s={0}
      />
      <Hexagon
        q={3}
        r={-2}
        s={0}
      />
      <Hexagon
        q={4}
        r={-2}
        s={0}
      />
      <Hexagon
        q={5}
        r={-2}
        s={0}
      />
      <Hexagon
        q={6}
        r={-2}
        s={0}
      />
      <Hexagon
        q={-6}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-5}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-4}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-3}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-2}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-1}
        r={-1}
        s={0}
      />
      <Hexagon
        q={0}
        r={-1}
        s={0}
      />
      <Hexagon
        q={1}
        r={-1}
        s={0}
      />
      <Hexagon
        q={2}
        r={-1}
        s={0}
      />
      <Hexagon
        q={3}
        r={-1}
        s={0}
      />
      <Hexagon
        q={4}
        r={-1}
        s={0}
      />
      <Hexagon
        q={5}
        r={-1}
        s={0}
      />
      <Hexagon
        q={-7}
        r={0}
        s={0}
      />
      <Hexagon
        q={-6}
        r={0}
        s={0}
      />
      <Hexagon
        q={-5}
        r={0}
        s={0}
      />
      <Hexagon
        q={-4}
        r={0}
        s={0}
      />
      <Hexagon
        q={-3}
        r={0}
        s={0}
      />
      <Hexagon
        q={-2}
        r={0}
        s={0}
      />
      <Hexagon
        q={-1}
        r={0}
        s={0}
      />
      <Hexagon
        q={0}
        r={0}
        s={0}
      />
      <Hexagon
        q={1}
        r={0}
        s={0}
      />
      <Hexagon
        q={2}
        r={0}
        s={0}
      />
      <Hexagon
        q={3}
        r={0}
        s={0}
      />
      <Hexagon
        q={4}
        r={0}
        s={0}
      />
      <Hexagon
        q={5}
        r={0}
        s={0}
      />
      <Hexagon
        q={-7}
        r={1}
        s={0}
      />
      <Hexagon
        q={-6}
        r={1}
        s={0}
      />
      <Hexagon
        q={-5}
        r={1}
        s={0}
      />
      <Hexagon
        q={-4}
        r={1}
        s={0}
      />
      <Hexagon
        q={-3}
        r={1}
        s={0}
      />
      <Hexagon
        q={-2}
        r={1}
        s={0}
      />
      <Hexagon
        q={-1}
        r={1}
        s={0}
      />
      <Hexagon
        q={0}
        r={1}
        s={0}
      />
      <Hexagon
        q={1}
        r={1}
        s={0}
      />
      <Hexagon
        q={2}
        r={1}
        s={0}
      />
      <Hexagon
        q={3}
        r={1}
        s={0}
      />
      <Hexagon
        q={4}
        r={1}
        s={0}
      />
      <Path
        end={
          Hex {
            "q": -2,
            "r": 0,
            "s": 1,
          }
        }
        start={
          Hex {
            "q": 0,
            "r": 0,
            "s": 0,
          }
        }
      />
    </Layout>
  </HexGrid>
</section>
`;
